lib/upgrade/migrator/migrate.go

```diff
@@ -655,6 +657,62 @@ func (m *migrationHelper) queryAndConvertV1ToV3Nodes(data *MigrationData) error
 	return nil
 }
 
+// ShouldMigrate checks version information and reports if migration is needed
+// and is possible.
+func (m *migrationHelper) ShouldMigrate() (bool, error) {
+	_, err := m.clientv3.ClusterInformation().Get(context.Background(), "default", options.GetOptions{})
+	if err == nil {
+		// Nothing to do ClusterInformation exists so data has already been migrated
+		return false, nil
+	} else if _, ok := err.(cerrors.ErrorResourceDoesNotExist); !ok {
+		// The error indicates a problem with accessing the resource
+		return false, err
+	} else {
```

Comment:
The way this is structured, you'll hit this branch if you have a resource does not exist error.

Reply:
general coding nit:  This last `else` is not required since in the previous if branches you `return` - so you can drop the else and get an extra tabs worth of room for the code.

Reply:
Don't we want to hit this for a resourceDoesNotExist error?

Reply:
d'oh  - for some reason I thought we then dereferenced the cluster info - but we don't ...   in my mind we did because that's what i think we need to do.  Sorry!